gwdetchar/io/datafind.py

```diff
@@ -198,6 +198,8 @@ def get_data(channel, start, end, frametype=None, source=None,
                 'Could not determine observatory from frametype')
         except (HTTPError, JSONDecodeError):  # frame files not found
             pass
+    if len(source) == 0: # reset if no frames found
+        source = None
     if isinstance(source, list) and isinstance(channel, (list, tuple)):
         channel = remove_missing_channels(channel, source)
     if source is not None:  # read from frame files
```

Comment:
@dethodav, this can be done more simply by just changing line 203:

```suggestion
     if source:  # read from frame files
```

This logic would proceed to read from a frame cache only if `source` is neither `None`, `0`, nor an empty iterable, which should clear up the issue you're seeing.

Reply:
I see the issue with my original commit, and this is definitely cleaner. However, there does still appear to be a needed change on line 201.